src/main/java/com/faforever/client/rankedmatch/MatchmakerMessage.java

```diff
@@ -12,13 +12,16 @@
   public static class MatchmakerQueue {
 
     private String queueName;
+    @SerializedName("queue_pop_time")
+    private String queuePopTime;
```

Comment:
Is this really a string? I assume it's a timestamp here? So Gson should be able to convert it into some java.date class? (I only know how to do it with Jackson)

Reply:
Well its a fromatted ISO datetime. Don't know if gson can do that automatically.

Reply:
Ok i just looked it up, GSON cannot deal with java.time by default, so we would have to use something like https://github.com/gkopff/gson-javatime-serialisers or use our own serializer. Don;t know if its worth it for one occurence.

Reply:
You can afaik configure gson when using a builder for it to configure how ro serialize specific classes.

Reply:
I think our gson has somthing that can deserialize dates....

Reply:
ok we did not but maybe we should set some TypeAdapter

Reply:
There is `LocalDate`, `LocalDateTime`, `OffsetDateTime`, `ZonedDateTime` and `Instant` in `java-time`. And there is no straight-forward 1 to 1 mapping between these times and ISO date formats. The whole formatter stuff exists because it the format needs to be specifically known before you can convert anything. So you would at least need to be able to specify the format for every `java.time` member. Not trivial and imo an overkill abstraction.

Reply:
Well ok